ui/components/app/signature-request/signature-request.component.js

```diff
@@ -221,8 +219,14 @@ export default class SignatureRequest extends PureComponent {
           .toBase(10)
           .toString();
 
-    const onSign = (event) => {
-      sign(event);
+    const onSign = async () => {
+      await resolvePendingApproval(id);
+      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+      if (this.props.mmiOnSignCallback) {
```

Comment:
What was happening before as the diff shows no previous MMI logic here?

Do we still need the code fencing if this is in an optional property?

Reply:
> What was happening before as the diff shows no previous MMI logic here?
- Yes MMI was actually currying sign function on container [here](https://github.com/MetaMask/metamask-extension/blob/546d8349e72cdde5e5a1ad4b02aba66f066fe2af/ui/components/app/signature-request/signature-request.container.js#L218) but we don't have sign methods passed into components anymore hence we just want `mmiOnSignCallback` to be called here.
> Do we still need the code fencing if this is in an optional property?
- I believe having code fencing still adds a small value here since this code will not be added into retail version.

Reply:
Btw this part will particularly will be reviewed by MMI, so is open to change soon.